backend/src/main/kotlin/org/loculus/backend/utils/EarliestReleaseDateFinder.kt

```diff
@@ -0,0 +1,74 @@
+package org.loculus.backend.utils
+
+import kotlinx.datetime.LocalDate
+import kotlinx.datetime.LocalDateTime
+import kotlinx.datetime.LocalTime
+import mu.KotlinLogging
+import org.loculus.backend.service.submission.RawProcessedData
+
+private val log = KotlinLogging.logger { }
+
+/**
+ * Calculate the earliest release date for rows of sequence entries given to it one by one.
+ * Assumes that rows are sorted: all accession entries are given in a block, and with ascending versions.
+ *
+ * The earliest release date of a sequence is the earliest date of:
+ * - the internal release date
+ * - any date from a given list of fields
+ * - the earliest release date from the previous version (if it exists)
+ */
+class EarliestReleaseDateFinder(private val fields: List<String>) {
+    private val earliestReleaseDateCache = mutableMapOf<String, LocalDateTime>()
```

Comment:
Instead of `String` this should probably be `AccessionVersion` for clarity?

Reply:
I think it's clear from context. If the PR would still be open I would be open to change it but I don't think it warrants a new PR.